packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -58,13 +61,70 @@ function readBlobContent(content: ISummaryBlob["content"]): unknown {
     return JSON.parse(json);
 }
 
+const flushPromises = async () => new Promise((resolve) => process.nextTick(resolve));
+
+let summarizer;
+
 // REVIEW: enable compat testing?
 describeNoCompat("Summaries", (getTestObjectProvider) => {
     let provider: ITestObjectProvider;
     beforeEach(() => {
         provider = getTestObjectProvider();
     });
 
+    it("On demand summaries", async () => {
+        const container = await createContainer(provider, { disableIsolatedChannels: false });
+        const defaultDataStore = await requestFluidObject<TestDataObject>(container, defaultDataStoreId);
+        const containerRuntime = defaultDataStore.getContext().containerRuntime as ContainerRuntime;
```

Comment:
this is an anti-pattern, that we only use in test.  a containerRuntime shouldn't expose itself, or it's context, the same goes for datastores. the right pattern is to issue a specially crafted request on the container which then returns a summarizer.  this is basically what the containerRuntime does internally to get its summarizer

Reply:
do we have a document or issue that describes the API here?

Reply:
The API is called out in https://github.com/microsoft/FluidFramework/issues/6260. Today, SummaryManager calls it to create the summarizer. The idea was to expose it to other callers. In the test, I wanted to spin up a summarizer on the fly, use it to summarize on demand, and then tear it down.

Reply:
The API I added to ContainerRuntime calls the existing method, which crafts the request that you're talking about.

Reply:
I'm not sure that this flow (of creating summarizer off existing container instance) is interesting. In that issue I've mentioned that
> One big item to consider in this design is how / when we can get to a point where server (controlling summarization process) could avoid having two container instances in order to produce summaries.

I.e. I think the direction we want to go is to be able to create summarizer instance directly, without having to have non-summarizer container. I.e. loader + URI + options -> summarizer.

Reply:
Right. The only reason ContainerRuntime.formRequestSummarizerFn() is used in this test is to give access to the loader. If the code has access to the proper loader (I'm not sure how that's done), then it can create a Summarizer on the fly without having a container instance.

Reply:
but the test has the loader. there is no need to access the ContainerRuntimes, its the same one (kind of) that the original container was loaded from

Reply:
also, when i've grabbed the summarizer i've done something like this, 
```
const container = await loader.resolve(containerReques);
  const request: IRequest = {
      headers: {
          [LoaderHeader.cache]: cache,
          [LoaderHeader.clientDetails]: {
              capabilities: { interactive: false },
              type: summarizerClientType,
          },
          [DriverHeader.summarizingClient]: summarizingClient,
          [LoaderHeader.reconnect]: reconnect,
          [LoaderHeader.sequenceNumber]: lastSequenceNumber,
      },
      url: "/_summarizer",
  };
const summarizer = await requestFluidObject<FluidObject<ISummarizer>>(container, request);
```

there is nothing special about the request off the loader the container runtime users. all request flow downward, so you can just as easily make the same request on the container you get from the loader.

Reply:
the above gets a container from the loader with all the properties of a summarizer, then request the summarizer itself from that container

Reply:
a bit of a rant, but i really dislike our e2e test api, it hides a ton of stuff. look at this test for and example of how to get a loader, then get your own containers off of it: https://github.com/microsoft/FluidFramework/blob/9a538cd4fe212324a052b9a26e70e3a84f4b110a/packages/test/test-end-to-end-tests/src/test/noDeltaStream.spec.ts#L47

Reply:
that test may go a bit farther than you need. it think the provider has a createLoader method you can use to get the loader, and then build up from there